main_test.go

```diff
@@ -28,9 +28,9 @@ func TestCheckGoVersion(t *testing.T) {
 		expectedErr error
 	}{
 		{minGoVersion, nil},
-		{minGoVersion + ".10", nil},
-		{"1.6.8", fmt.Errorf("Minio is not compiled by Go >= 1.7.  Please recompile accordingly.")},
-		{"0.1", fmt.Errorf("Minio is not compiled by Go >= 1.7.  Please recompile accordingly.")},
+		{"1.6.8", fmt.Errorf("Minio is not compiled by Go >= 1.8.3.  Please recompile accordingly")},
+		{"1.5", fmt.Errorf("Minio is not compiled by Go >= 1.8.3.  Please recompile accordingly")},
```

Comment:
> I couldn't able to understand why you removed minGoVersion + ".10" positive case.

Actually because this was failing, 1.8.3.10 is considered malformed version and we are requiring 1.8.3 as minimal version.